app/logic/emailHandler.py

```diff
@@ -133,22 +134,36 @@ def retrieve_and_modify_email_template(self):
         return (template_id, subject, new_body)
 
     def attach_attachments(self):
-        # TODO for later
-        # retrieve attachments, attach it to the email
-        # Q: how would this work?
-        pass
+        """
+        This creates the directory/path for the object from the "Choose File" input in the emailModal.html file.
+        Additionally, it saves the attachment in the app/static/files/attachments/ directory.
+        :returns: directory path for attachment
+        """
+        attachmentObject = request.files['attachmentObject']
+        attachmentName = attachmentObject.filename
+        attachment_path = 'app/static/files/attachments'
+        try:  # tries to create the attachment_path directory and passes if already created
+            os.mkdir(attachment_path)
+        except:
+            pass
+        attachmentDirectory = os.path.join(attachment_path, attachmentName) # joins attachmnet_path with attachment name to create directory
+        attachmentObject.save(attachmentDirectory) # saves attachment in directory
+        return attachmentDirectory
+
 
     def store_sent_email(self, subject, template_id):
         """ Stores sent email in the email log """
         date_sent = datetime.now()
+
         EmailLog.create(
-            event=self.event.id,
-            subject=subject,
-            templateUsed=template_id,
-            recipientsCategory=self.recipients_category,
-            recipients=", ".join(recipient.email for recipient in self.recipients),
-            dateSent=date_sent,
-            sender=self.sender)
+            event = self.event.id,
+            subject = subject,
+            templateUsed = template_id,
+            recipientsCategory = self.recipients_category,
+            recipients = ", ".join(recipient.email for recipient in self.recipients),
+            dateSent = date_sent,
+            sender = self.sender,
+            attachmentDirectory = self.attach_attachments())

```

Comment:
This may be something that is addressed later when we want to figure out if and when we are saving files sent in the database. The attach attachments function currently is creating the path as well as saving the attachments to this path that is created. If we do want to save attachments this can be renamed to "getAttachementDirectory"